src/framework/auth/services/interceptors/jwt-interceptor.ts

```diff
@@ -13,24 +13,44 @@ export class NbAuthJWTInterceptor implements HttpInterceptor {
   }
 
   intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-
-    return this.authService.getToken()
-      .pipe(
-        switchMap((token: NbAuthJWTToken) => {
-          if (token.isValid()) {
-            const JWT = `Bearer ${token.getValue()}`;
-            req = req.clone({
-              setHeaders: {
-                Authorization: JWT,
-              },
-            });
-          }
-          return next.handle(req);
-        }),
-      );
+    // do not intercept request whose urls are filtered by the injected filter
+      if (!this.filter(req)) {
+        return this.authService.isAuthenticatedOrRefresh()
+          .pipe(
+            switchMap(authenticated => {
+              if (authenticated) {
+                  return this.authService.getToken().pipe(
+                    switchMap( (token: NbAuthToken) => {
```

Comment:
space

Reply:
@alain-charles just one last formatting issue here :)
`switchMap( (token: NbAuthToken) => {` -> `switchMap((token: NbAuthToken) => {`

Reply:
I run npm lint before I push but it seems my config is not as restrictive as yours .
Which one do you use ?

Reply:
It looks like the project lint config isn't the best one. Will look into it.